ui-tests/tests/lite.spec.ts

```diff
@@ -0,0 +1,88 @@
+import { expect, test, galata } from '@jupyterlab/galata';
+
+test.use({ autoGoto: false });
+
+test.describe('UI Test', () => {
+  const fileList = ['test.jcad', '3M_CONNECTOR.STEP', 'fan.stl'];
+  let errors = 0;
+  test.beforeEach(async ({ page }) => {
+    page.setViewportSize({ width: 1920, height: 1080 });
+    page.on('console', message => {
+      if (message.type() === 'error') {
+        console.log('ERROR MSG', message.text());
+        errors += 1;
+      }
+    });
+  });
+
+  test.afterEach(async ({ page }) => {
+    errors = 0;
+  });
+
+  for (const file of fileList) {
+    test(`Should be able to render ${file} without error`, async ({
+      browser
+    }) => {
+      const context = await browser.newContext();
+      const page = await context.newPage();
+
+      await page.goto(`lab/index.html?path=${file}`, {
+        waitUntil: 'domcontentloaded'
+      });
+      console.log('FILE LOADED');
+
+      await page.locator('div.jpcad-Spinner').waitFor({ state: 'hidden' });
+
+      if (await page.getByRole('button', { name: 'Ok' }).isVisible()) {
+        await page.getByRole('button', { name: 'Ok' }).click();
+      }
+
+      const main = await page.waitForSelector('.jp-MainAreaWidget', {
+        state: 'visible'
+      });
+
+      console.log('hurrayyyy', main);
+      await page.waitForTimeout(10000);
```

Comment:
Have you explored using https://playwright.dev/docs/api/class-pageassertions#page-assertions-to-have-screenshot-1 instead?

Reply:
Yes, the limitation I came across was that it is a property of page and then i cannot take screenshot of the part i need but whole page and on top of that using this doesn't seem to solve the issue of pre-loaded screenshot